pkg/controller/nodemaintenance/taint.go

```diff
@@ -40,29 +40,31 @@ func AddOrRemoveTaint(clientset kubernetes.Interface, node *corev1.Node, add boo
 
 	log.Info(fmt.Sprintf("Applying %s taint %s on Node: %s", taint.Key, taintStr, node.Name))
 
-	if !updated {
-		log.Error(err, fmt.Sprintf("%s taint %s was not applied on Node: %s", taint.Key, taintStr, node.Name))
-	}
-
-	if err != nil {
-		return err
-	}
-
-	newData, err := json.Marshal(freshNode)
 	if err != nil {
 		return err
 	}
 
-	patchBytes, patchErr := strategicpatch.CreateTwoWayMergePatch(oldData, newData, node)
-	if patchErr == nil {
-		_, err = client.Patch(node.Name, types.StrategicMergePatchType, patchBytes)
+	if !updated {
+		log.Info(fmt.Sprintf("%s taint %s was not applied on Node: %s", taint.Key, taintStr, node.Name))
 	} else {
-		_, err = client.Update(node)
-	}
-
-	if patchErr != nil {
-		log.Error(err, fmt.Sprintf("%s taint %s was not applied on Node: %s", taint.Key, taintStr, node.Name))
+		newData, err := json.Marshal(freshNode)
+		if err != nil {
+			return err
+		}
+
+		patchBytes, patchErr := strategicpatch.CreateTwoWayMergePatch(oldData, newData, node)
+		if patchErr == nil {
+			_, err = client.Patch(node.Name, types.StrategicMergePatchType, patchBytes)
```

Comment:
I just checked, tains don't have a "merge key" for the strategic merge patch. Tains have a "Key" field, but it does not act as key for the patch operation. Can you change the patch type to json patch?

This will require a patch operation which consists of a `test` and a `replace` operation. Here an example: https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-controller/watch/vmi.go#L370.

If the list can also be `nil`, you may have to check for  that too and do an `add` if it is nil.